File: app/models/entity/report.go
package entity

import (
	"time"

	"github.com/Spicy-Bush/fider-tarkov-community/app/models/enum"
)

type Report struct {
	ID             int               `json:"id"`
	ReportedType   enum.ReportType   `json:"reportedType"`
	ReportedID     int               `json:"reportedId"`
	Reason         string            `json:"reason"`
	Details        string            `json:"details,omitempty"`
	Status         enum.ReportStatus `json:"status"`
	CreatedAt      time.Time         `json:"createdAt"`
	Reporter       *User             `json:"reporter"`
	AssignedTo     *User             `json:"assignedTo,omitempty"`
	AssignedAt     *time.Time        `json:"assignedAt,omitempty"`
	ResolvedAt     *time.Time        `json:"resolvedAt,omitempty"`
	ResolvedBy     *User             `json:"resolvedBy,omitempty"`
	ResolutionNote string            `json:"resolutionNote,omitempty"`
	PostNumber     int               `json:"postNumber,omitempty"`
	PostSlug       string            `json:"postSlug,omitempty"`
}

type ReportReason struct {
	ID          int    `json:"id"`
	Slug        string `json:"slug"`
	Title       string `json:"title"`
	Description string `json:"description,omitempty"`
	SortOrder   int    `json:"sortOrder"`
	IsActive    bool   `json:"isActive"`
}

File: public/models/report.ts
import { User } from "./identity"

export type ReportType = "post" | "comment"

export type ReportStatus = "pending" | "in_review" | "resolved" | "dismissed"

export interface Report {
  id: number
  reportedType: ReportType
  reportedId: number
  reason: string
  details?: string
  status: ReportStatus
  createdAt: string
  reporter: User
  assignedTo?: User
  assignedAt?: string
  resolvedAt?: string
  resolvedBy?: User
  resolutionNote?: string
  postNumber?: number
  postSlug?: string
}

export interface ReportReason {
  id: number
  slug: string
  title: string
  description?: string
  sortOrder: number
  isActive: boolean
}

export const getReportTypeLabel = (type: ReportType | string): string => {
  switch (type) {
    case "post":
      return "Post"
    case "comment":
      return "Comment"
    default:
      return type
  }
}

